src/MetaTable.tsx

```diff
@@ -1,65 +1,139 @@
-import React from 'react';
+import React, {useCallback, useEffect, useState} from 'react';
+import {mergeData, renameColumns} from "./table-data-transform";
+import {DataDeep, Datum} from "gosling.js/dist/src/core/gosling.schema";
+import {GoslingRef} from "gosling.js";
+
+export type MetaTableSpec = {
+    type: "table",
+    // TODO: allow custom data specification for metatable
+    data?: DataDeep;
+    dataTransform: tableDataTransform[];
+    genomicColumns: [string,string] | [string];
+    columns: string[];
+}
 
 interface MetaTableProps {
```

Comment:
You could reuse `MetaTableSpec` instead of redefining `dataTransform`, `genomicColumns`, and `columns`:

```ts
interface MetaTableProps extends Omit<MetaTableSpec, 'type' | 'data'> {
  ...
}
```